Timeout whilst creating an address's configmap reported to client with 500 NullPointerException

**Describe the bug**
During a test run, the test initiated a creation of an address.  The POST made it to the API server but the resulting request to create the configmap ended with a HTTP2 stream timeout.  The underlying reason why the configmap creation timed out is not known.  (The timeout is 10 second timeout and this is a default coming from `okhttp3.OkHttpClient.Builder#Builder()`)

However, the api-server's handling of the error lead to a NPE.

agent log:

```
2019-10-07T00:32:17.359Z agent INFO  creating address definition {"address":"topic-via-web-0","type":"topic","plan":"standard-small-topic","topic":null}
2019-10-07T00:32:27.377Z agent INFO  POST /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 500
```

api-server log:

```
2019-10-07T00:32:27.373Z ERROR [HttpAddressServiceBase] Error creating address
io.fabric8.kubernetes.client.KubernetesClientException: Operation: [create]  for kind: [ConfigMap]  with name: [enmasse-infra.standard-shared-0.topic-via-web-0.ae7c2b33-6a38-4442-b45d-5f3489768c7e]  in namespace: [enmasse-infra]  failed.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:72)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:384)
	at io.enmasse.k8s.api.ConfigMapAddressApi.createAddress(ConfigMapAddressApi.java:144)
	at io.enmasse.api.v1.AddressApiHelper.createAddress(AddressApiHelper.java:119)
	at io.enmasse.api.v1.http.HttpAddressServiceBase.lambda$internalCreateAddress$4(HttpAddressServiceBase.java:116)
	at io.enmasse.api.v1.http.HttpAddressServiceBase.doRequest(HttpAddressServiceBase.java:62)
	at io.enmasse.api.v1.http.HttpAddressServiceBase.internalCreateAddress(HttpAddressServiceBase.java:114)
	at io.enmasse.api.v1.http.HttpAddressService.createAddress(HttpAddressService.java:61)
	at jdk.internal.reflect.GeneratedMethodAccessor183.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:509)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:399)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$0(ResourceMethodInvoker.java:363)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:365)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:337)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:310)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:443)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:233)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:139)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:142)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:219)
	at org.jboss.resteasy.plugins.server.vertx.RequestDispatcher.service(RequestDispatcher.java:93)
	at org.jboss.resteasy.plugins.server.vertx.VertxRequestHandler$1.handle(VertxRequestHandler.java:70)
	at org.jboss.resteasy.plugins.server.vertx.VertxRequestHandler$1.handle(VertxRequestHandler.java:52)
	at io.vertx.core.http.HttpServerRequest.lambda$bodyHandler$0(HttpServerRequest.java:214)
	at io.vertx.core.http.impl.HttpServerRequestImpl.onEnd(HttpServerRequestImpl.java:529)
	at io.vertx.core.http.impl.HttpServerRequestImpl.handleEnd(HttpServerRequestImpl.java:515)
	at io.vertx.core.http.impl.Http1xServerConnection.handleEnd(Http1xServerConnection.java:172)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:134)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.WorkerContext.lambda$wrapTask$0(WorkerContext.java:34)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.SocketTimeoutException: timeout
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.newTimeoutException(Http2Stream.java:656)
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.exitAndThrowIfTimedOut(Http2Stream.java:664)
	at okhttp3.internal.http2.Http2Stream.takeHeaders(Http2Stream.java:153)
	at okhttp3.internal.http2.Http2Codec.readResponseHeaders(Http2Codec.java:131)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:119)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.lambda$createHttpClient$1(HttpClientUtils.java:100)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:254)
	at okhttp3.RealCall.execute(RealCall.java:92)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:379)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:227)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:801)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:377)
	... 38 common frames omitted
2019-10-07T00:32:27.376Z ERROR [i18n] RESTEASY019525: Unexpected
java.lang.NullPointerException: null
	at io.enmasse.api.common.DefaultExceptionMapper.toResponse(DefaultExceptionMapper.java:32)
	at io.enmasse.api.common.DefaultExceptionMapper.toResponse(DefaultExceptionMapper.java:22)
	at org.jboss.resteasy.core.ExceptionHandler.executeExceptionMapper(ExceptionHandler.java:102)
	at org.jboss.resteasy.core.ExceptionHandler.unwrapException(ExceptionHandler.java:131)
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:76)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:222)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:197)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:459)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:233)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:139)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:142)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:219)
	at org.jboss.resteasy.plugins.server.vertx.RequestDispatcher.service(RequestDispatcher.java:93)
	at org.jboss.resteasy.plugins.server.vertx.VertxRequestHandler$1.handle(VertxRequestHandler.java:70)
	at org.jboss.resteasy.plugins.server.vertx.VertxRequestHandler$1.handle(VertxRequestHandler.java:52)
	at io.vertx.core.http.HttpServerRequest.lambda$bodyHandler$0(HttpServerRequest.java:214)
	at io.vertx.core.http.impl.HttpServerRequestImpl.onEnd(HttpServerRequestImpl.java:529)
	at io.vertx.core.http.impl.HttpServerRequestImpl.handleEnd(HttpServerRequestImpl.java:515)
	at io.vertx.core.http.impl.Http1xServerConnection.handleEnd(Http1xServerConnection.java:172)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:134)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.WorkerContext.lambda$wrapTask$0(WorkerContext.java:34)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-10-07T00:32:27.376Z INFO  [HTTPServer] Request GET /apis/enmasse.io/v1beta1
```

**To Reproduce**
N/K

**Expected behavior**
The api-server ought to be able to handle Kubernetes timeouts gracefully.  503  Service Unavailable would be a more appropriate error response.

--- a/api-common/src/main/java/io/enmasse/api/common/DefaultExceptionMapper.java
+++ b/api-common/src/main/java/io/enmasse/api/common/DefaultExceptionMapper.java
@@ -19,6 +19,8 @@ import io.fabric8.kubernetes.client.KubernetesClientException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.net.SocketTimeoutException;
+
 @Provider
 public class DefaultExceptionMapper implements ExceptionMapper<Exception> {
     protected static final Logger log = LoggerFactory.getLogger(DefaultExceptionMapper.class.getName());
@@ -29,7 +31,15 @@ public class DefaultExceptionMapper implements ExceptionMapper<Exception> {
         if (exception instanceof WebApplicationException) {
             statusCode = ((WebApplicationException) exception).getResponse().getStatus();
         } else if (exception instanceof KubernetesClientException) {
-            statusCode = ((KubernetesClientException) exception).getStatus().getCode();
+            KubernetesClientException kce = (KubernetesClientException) exception;
+            if (kce.getStatus() != null) {
+                statusCode = kce.getStatus().getCode();
+            } else if (kce.getCause() instanceof SocketTimeoutException) {
+                statusCode = Response.Status.SERVICE_UNAVAILABLE.getStatusCode();
+            } else {
+                statusCode = Response.Status.INTERNAL_SERVER_ERROR.getStatusCode();
+            }
+
         } else if (exception instanceof UnresolvedAddressException || exception instanceof JsonProcessingException || exception instanceof UnresolvedAddressSpaceException || exception instanceof ValidationException || exception instanceof UserValidationFailedException) {
             statusCode = Response.Status.BAD_REQUEST.getStatusCode();
         } else if (exception instanceof KeycloakUnavailableException) {

--- a/api-common/src/test/java/io/enmasse/api/common/DefaultExceptionMapperTest.java
+++ b/api-common/src/test/java/io/enmasse/api/common/DefaultExceptionMapperTest.java
@@ -11,6 +11,8 @@ import org.junit.jupiter.api.Test;
 import javax.ws.rs.WebApplicationException;
 import javax.ws.rs.core.Response;
 
+import java.net.SocketTimeoutException;
+
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
@@ -46,6 +48,38 @@ public class DefaultExceptionMapperTest {
         assertTrue(response.getEntity() instanceof Status);
         Status responseEntity = (Status) response.getEntity();
         assertEquals("Unprocessable Entity", responseEntity.getReason());
+        assertEquals(message, responseEntity.getMessage());
+    }
+
+    @Test
+    public void testToResponseStatus500() {
+        int code = 500;
+        Response.Status status = Response.Status.fromStatusCode(code);
+        String message = "Some error message";
+        KubernetesClientException kubernetesClientException = new KubernetesClientException(message, new NullPointerException("something's gone bad!"));
+
+        Response response = new DefaultExceptionMapper().toResponse(kubernetesClientException);
+        assertEquals(code, response.getStatus());
+        assertEquals(status.getReasonPhrase(), response.getStatusInfo().getReasonPhrase());
+        assertTrue(response.getEntity() instanceof Status);
+        Status responseEntity = (Status) response.getEntity();
+        assertEquals("Internal Server Error", responseEntity.getReason());
+        assertEquals(message, responseEntity.getMessage());
+    }
+
+    @Test
+    public void testToResponseStatus503() {
+        int code = 503;
+        Response.Status status = Response.Status.fromStatusCode(code);
+        String message = "Some error message";
+        KubernetesClientException kubernetesClientException = new KubernetesClientException(message, new SocketTimeoutException());
+
+        Response response = new DefaultExceptionMapper().toResponse(kubernetesClientException);
+        assertEquals(code, response.getStatus());
+        assertEquals(status.getReasonPhrase(), response.getStatusInfo().getReasonPhrase());
+        assertTrue(response.getEntity() instanceof Status);
+        Status responseEntity = (Status) response.getEntity();
+        assertEquals("Service Unavailable", responseEntity.getReason());
         assertEquals(message, responseEntity.getMessage());
     }
 

